### ISSUE TITLE
FlexAttention: Support for post_mod

### ISSUE THREAD

**Issue by EpicUsaMan** [pytorch/pytorch#153221]
### 🚀 The feature, motivation and pitch

post_mod is essential to implement such types of attention like MultiTokenAttention

Current score_mod gives ability to implement pre-softmax variant of MTA, but don't allow to implement post-softmax or head-mix variant

For more flexibility post_mod could return not only modified post-softmax scores, but also modified V, but it's not required by MTA paper

### Additional context

More details: https://arxiv.org/pdf/2504.00927

**Comment by EpicUsaMan** [comment]
Any news? @jbschlosser
